nomad/structs/structs.go

```diff
@@ -1135,6 +1135,11 @@ func (r *RestartPolicy) Validate() error {
 		return fmt.Errorf("Unsupported restart mode: %q", r.Mode)
 	}
 
+	// Check for ambiguous/confusing settings
+	if r.Attempts == 0 && r.Mode != RestartPolicyModeFail {
+		return fmt.Errorf("Restart policy %q with %d attempts is ambiguous", r.Mode, r.Attempts)
+	}
+
 	if r.Interval == 0 {
 		return nil
```

Comment:
Thinking about this a bit more, a zero interval should be OK. For example I want to provide a delay between restarts, but I don't want to limit the number of times a restart can happen for a given time period. It will also be common to have a zero interval for the zero attempts case (user doesn't need to specify it in that case). Let me know if this isn't the behavior we want.